mautic/mautic
# Fix remove from campaign action prevent loop

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
If you use remove from campaign action, then with combination `Allow contacts to restart the campaign`  it make loop event triggering. As I understand well `date_last_exited` column in `campaign_leads` table should be use If contact is removed from campaign by segment/any source. Then If we remove contact from campaign by action, It should be manually.




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Create segment with filter 
3. After segment rebuild you should there at least one contact
4. Create campaign  and enable  `Allow contacts to restart the campaign` and add source of created segment
5.  Add remove from campaign action for this campaign
6. Rebuild campaign
7. Then trigger campaign
8. Rebuild campaign
9. Then trigger again
10. Rebuild campaign
11. Then trigger
12. This make loop
